packages/core/test/types.test.ts

```diff
@@ -440,7 +440,7 @@ describe('context', () => {
   });
 
   it('should work with generic context', () => {
-    function createMachineWithExtras<TContext>(
+    function createMachineWithExtras<TContext extends {}>(
```

Comment:
It probably was a slight bug in the past and it got uncovered by the new, more conservative, rules around unconstrained generics. To fix some inference problems we have introduced `LowInfer` here:
https://github.com/statelyai/xstate/blob/9f78588ead2361205e1c8318930d3eab392b33df/packages/core/src/types.ts#L983

We didn't really want to constrain the `TContext` to be a subtype of `{}` but that's the effect with the current version of TS. In practice, this shouldn't be a big problem because we only accept object contexts at runtime. Some people might be forced to add `extends {}` in their generic declarations, just like I've done here, but that's just about it.